backend-models/service-tests/showDuggaservice-test.txt

```diff
@@ -0,0 +1,393 @@
+showDuggaservice.php
+
+—------------------------------------------------------------------------------------------------------------------------
+TEST #1 - 100-105
+Get active users
+
+***
+Pre-req: -
+***
+Send parameters inputs to showDuggaservice.php
+    send
+    {
+        $opt = “UPDATEAU”
+        $hash = hash
+
+    }
+
+    SQL-query: SELECT active_users FROM groupdugga WHERE hash=hash
+    This should show all active users
+
+***
+Gather service output
+***
+    Save the values of the echoed array as a JSON (at the end of the file. echo json_encode($array);)
+    This is the expected output for the micro service.
+
+—------------------------------------------------------------------------------------------------------------------
+TEST #2 - 106-110
+Create active users
+
+***
+Pre-req: $active == NULL
+***
+
+Send parameters inputs to showDuggaservice.php
+    send
+    {
+        $opt = “UPDATEAU”
+        $hash = “tj7dh2nb”
+        $AUtoken = 999
+    }
+
+    SQL-query: INSERT INTO groupdugga(hash,active_users) VALUES(:hash,:AUtoken)
+***
+* Get newly added active users
+***
+	SQL-query: SELECT MAX(active_users) FROM groupdugga; // Save to variable $active_users this will identify the last active user that was created (it has the highest AUtoken)
+
+***
+Gather service output
+***
+    Save the values of the echoed array as a JSON (at the end of the file. echo json_encode($array);)
+    This is the expected output for the micro service.
+
+***
+* Remove inserted row from DB, using hash
+***
+
+	SQL-query DELETE FROM groupdugga WHERE hash = $hash;
+
+—------------------------------------------------------------------------------------------------------------------
+
+—------------------------------------------------------------------------------------------------------------------
+TEST #3 - 111-117
+update active users
+
+***
+Pre-req: $active != NULL
+***
+
+***
+Pre-req variables
+***
+$pre-values =
+    {
+        $hash = “hjk4ert6”
+        $AUtoken = 52
+    }
+
+    SQL-query: INSERT INTO groupdugga(hash,active_users) VALUES($pre-values[0], $pre-values[1])
```

Comment:
This insert is as you probably understand by now, not needed. 
But this might be as good a time as ever to mention that, if you use values from an array, you also need to specify what those values are.

im presuming that 
$pre-values[0] = hjk4ert6
$pre-values[1] = 52
but this can be misunderstood by some. 

it should follow this standard
```
$preValuesVariant = {   $quizID,
                        2,
                        0,
                        '{&quot;type&quot;:&quot;md&quot;,&quot;filelink&quot;:&quot;md&quot;,&quot;gType&quot;:&quot;&quot;,&quot;diagram_File&quot;:&quot;Empty canvas&quot;,&quot;diagram_type&quot;:{&quot;ER&quot;:true,&quot;UML&quot;:false,&quot;IE&quot;:false},&quot;extraparam&quot;:&quot;&quot;,&quot;notes&quot;:&quot;&quot;,&quot;submissions&quot;:[{&quot;type&quot;:&quot;pdf&quot;,&quot;fieldname&quot;:&quot;&quot;,&quot;instruction&quot;:&quot;&quot;}],&quot;errorActive&quot;:false}',
                        "some text" };

SQL-Query:  INSERT INTO variant(quizID,creator,disabled,param,variantanswer) 
            VALUES (preValuesVariant[0], preValuesVariant[1], preValuesVariant[2], preValuesVariant[3], preValuesVariant[4]);
```